src/main/scala/com/databricks/labs/overwatch/api/ApiCallV2.scala

```diff
@@ -343,9 +343,9 @@ class ApiCallV2(apiEnv: ApiEnv) extends SparkSessionWrapper {
       execute()
     } else {
       if(writeTraceApiFlag()){
-        PipelineFunctions.writeMicroBatchToTempLocation(successTempPath.get, apiMeta.enrichAPIResponse(response,jsonQuery,queryMap))
+        PipelineFunctions.writeMicroBatchToTempLocation(successTempPath.get, apiMeta.enrichAPIResponse(response.body,response.code,jsonQuery,queryMap))
       }
-      throw new ApiCallFailure(response, buildGenericErrorMessage, debugFlag = false)
+      throw new ApiCallFailure(response, buildGenericErrorMessage, responseWithMeta = apiMeta.enrichAPIResponse(jsonQueryToApiErrorDetail(response.body),response.code, jsonQuery, queryMap) , debugFlag = false)
```

Comment:
For readability, you should do function calls for param evaluation separately

Reply:
Implemented